File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Sizing/SizingRoles.swift
import Foundation

public enum SizingRoles {
    public enum Icon {
        case xs
        case s
        case m
        case l
        case xl
        case xxl

        public var rawValue: CGFloat {
            switch self {
            case .xs: SizingPrimitives.size12.rawValue
            case .s: SizingPrimitives.size16.rawValue
            case .m: SizingPrimitives.size24.rawValue
            case .l: SizingPrimitives.size32.rawValue
            case .xl: SizingPrimitives.size44.rawValue
            case .xxl: SizingPrimitives.size108.rawValue
            }
        }
    }

    public enum Avatar {
        case s
        case m
        case l

        public var rawValue: CGFloat {
            switch self {
            case .s: SizingPrimitives.size40.rawValue
            case .m: SizingPrimitives.size60.rawValue
            case .l: SizingPrimitives.size88.rawValue
            }
        }
    }

    public enum HorizontalRule {
        case xs
        case s

        public var rawValue: CGFloat {
            switch self {
            case .xs: SizingPrimitives.size1.rawValue
            case .s: SizingPrimitives.size2.rawValue
            }
        }
    }

    public enum InteractionHeight {
        case compact
        case `default`
        case comfort

        public var rawValue: CGFloat {
            switch self {
            case .compact: SizingPrimitives.size32.rawValue
            case .default: SizingPrimitives.size48.rawValue
            case .comfort: SizingPrimitives.size56.rawValue
            }
        }
    }

    public enum GridCell {
        public static let minWidth: CGFloat = SizingPrimitives.size128.rawValue
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/sizing/SizingRoles.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.sizing

import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.atomic._primitives.sizing.SizingPrimitives

object SizingRoles {
    enum class Icon(
        val dp: Dp,
    ) {
        XS(SizingPrimitives.Size12),
        S(SizingPrimitives.Size16),
        M(SizingPrimitives.Size24),
        L(SizingPrimitives.Size32),
        XL(SizingPrimitives.Size44),
        XXL(SizingPrimitives.Size108),
    }

    enum class Avatar(
        val dp: Dp,
    ) {
        S(SizingPrimitives.Size40),
        M(SizingPrimitives.Size60),
        L(SizingPrimitives.Size88),
    }

    enum class HorizontalRule(
        val dp: Dp,
    ) {
        XS(SizingPrimitives.Size1),
        S(SizingPrimitives.Size2),
    }

    enum class InteractionHeight(
        val dp: Dp,
    ) {
        Compact(SizingPrimitives.Size32),
        Default(SizingPrimitives.Size48),
        Comfort(SizingPrimitives.Size56),
    }

    object GridCell {
        val MinWidth: Dp = SizingPrimitives.Size128
    }
}
